include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -961,135 +970,449 @@ void reduceSumScalingVector(common::Pipeline<comm::Communicator>& col_task_chain
-      // The element distance between the current tile column and the initial tile column in the global
-      // matrix tile grid
-      const SizeType j_gl_subm_el = dist.globalTileElementDistance<Coord::Col>(i_begin, j_gl_tile);
-
-      // Skip columns that are in the deflation zone
-      if (j_gl_subm_el >= k)
-        break;
-
-      // Iterate over the elements of the column tile
-      const SizeType ncols = std::min(dist.tileSize<Coord::Col>(j_gl_tile), k - j_gl_subm_el);
-      for (SizeType j_tile_el = 0; j_tile_el < ncols; ++j_tile_el) {
-        // The global submatrix column
-        const SizeType j_gl_el = j_gl_subm_el + j_tile_el;
-
-        // Solve the deflated rank-1 problem
-        T& eigenval = eval_ptr[to_sizet(j_gl_el)];
-        lapack::laed4(static_cast<int>(k), static_cast<int>(j_gl_el), d_ptr, z_ptr, delta_ptr, rho,
-                      &eigenval);
-
-        // Iterate over the rows of the local submatrix tile grid and copy the parts from delta stored on this rank.
-        for (SizeType i_loc_subm_tile = 0; i_loc_subm_tile < sz_loc_tiles.rows(); ++i_loc_subm_tile) {
-          const SizeType i_subm_evec_arr = i_loc_subm_tile + j_loc_subm_tile * sz_loc_tiles.rows();
-          auto& evec_tile = evec_tile_arr[to_sizet(i_subm_evec_arr)];
-          // The tile row in the local matrix tile grid
-          const SizeType i_loc_tile = idx_loc_begin.row() + i_loc_subm_tile;
-          // The tile row in the global matrix tile grid
-          const SizeType i_gl_tile = dist.globalTileFromLocalTile<Coord::Row>(i_loc_tile);
-          // The element distance between the current tile row and the initial tile row in the
-          // global matrix tile grid
-          const SizeType i_gl_subm_el = dist.globalTileElementDistance<Coord::Row>(i_begin, i_gl_tile);
-
-          // The tile row in the global submatrix tile grid
-          const SizeType i_subm_i2_arr = dist.globalTileFromLocalTile<Coord::Row>(i_loc_tile) - i_begin;
-          auto& i2_tile = i2_tile_arr[to_sizet(i_subm_i2_arr)];
-
-          const SizeType nrows = std::min(dist.tileSize<Coord::Row>(i_gl_tile), n - i_gl_subm_el);
-          for (int i = 0; i < nrows; ++i) {
-            const SizeType ii = i2_tile({i, 0});
-            if (ii < k)
-              evec_tile({i, j_tile_el}) = delta_ptr[ii];
+
+  TileCollector tc{i_begin, i_end};
+
+  const SizeType n = problemSize(i_begin, i_end, dist);
+
+  const SizeType m_subm_el_lc = [=]() {
+    const auto i_loc_being = ij_begin_lc.row();
+    const auto i_loc_end = ij_begin_lc.row() + sz_loc_tiles.rows();
+    return dist.localElementDistanceFromLocalTile<Coord::Row>(i_loc_being, i_loc_end);
+  }();
+
+  const SizeType n_subm_el_lc = [=]() {
+    const auto i_loc_being = ij_begin_lc.col();
+    const auto i_loc_end = ij_begin_lc.col() + sz_loc_tiles.cols();
+    return dist.localElementDistanceFromLocalTile<Coord::Col>(i_loc_being, i_loc_end);
```

Comment:
Sorry, one trivial fix that I forgot to comment about:
```suggestion
    const auto i_loc_begin = ij_begin_lc.row();
    const auto i_loc_end = ij_begin_lc.row() + sz_loc_tiles.rows();
    return dist.localElementDistanceFromLocalTile<Coord::Row>(i_loc_begin, i_loc_end);
  }();

  const SizeType n_subm_el_lc = [=]() {
    const auto i_loc_begin = ij_begin_lc.col();
    const auto i_loc_end = ij_begin_lc.col() + sz_loc_tiles.cols();
    return dist.localElementDistanceFromLocalTile<Coord::Col>(i_loc_begin, i_loc_end);
```